src/RectangleA.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Maman 12 - Class RectangleA.
+ * The RectangleA class represents a 2D axis aligned integral rectangles.
+ *
+ * @author Noa Amir
+ * @version 2022a
+ */
+
+public class RectangleA {
+    // 1. instance variables
+    private int _width;
+    private int _height;
+    private Point _pointSW;
+
+    // 2. constructors
+
+    /**
+     * First constructor for objects of class RectangleA Constructs a new rectangle with the specified width, height, and it's south-west corner is (0,0)
+     *
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleA(int w, int h) {
+        _pointSW = new Point(0, 0);
+
+        if (w < 1)
+            _width = 1; // TODO: Should I do final DEFAULT_WIDTH = 1
```

Comment:
Done.
Please see there are 3 uses in specific numbers (In both rectangles)- 

1. check that w, h are POSITIVE. 
There, I left the number 1 as is (w < 1, h < 1)
(Should use another constant like MIN_WIDTH? Don't want to overkill...)

2. Defaulting w, h to default number if they are not positive.
There, I added the constants :
    private final int DEFAULT_WIDTH = 1;
    private final int DEFAULT_HEIGHT = 1;
So if w, h are not positive - defaulting to them. If it will change in the future to 5 - can change the constants.

3. Set w, h - Set a new value to w\h, only if it's positive. There I also left the number as is: 
w > 0 , h > 0
(Will be right to use the same MIN_WIDTH final here if I'll create it?)

Let me know what you think